src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -16,9 +17,89 @@ public class ResourceNotificationService(ILogger<ResourceNotificationService> lo
     // Resource state is keyed by the resource and the unique name of the resource. This could be the name of the resource, or a replica ID.
     private readonly ConcurrentDictionary<(IResource, string), ResourceNotificationState> _resourceNotificationStates = new();
     private readonly ILogger<ResourceNotificationService> _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    private readonly CancellationToken _applicationStopping;
 
     private Action<ResourceEvent>? OnResourceUpdated { get; set; }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="ResourceNotificationService"/>.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="hostApplicationLifetime">The host application lifetime.</param>
+    public ResourceNotificationService(ILogger<ResourceNotificationService> logger, IHostApplicationLifetime hostApplicationLifetime)
```

Comment:
Do we do anything with the existing constructor when we add a new one with more parameters, e.g. obsolete it?

Reply:
Yeah ... obsolete it.